### PR TITLE
Fix framebuffers with multiple color slots

### PR DESCRIPTION
**Pull request by iwikal**
Not actually solved! This is just my progress so far. I don't really know where to go from here though, it still seems to bind them all to the same location. I did notice a change in behavior at this commit 58fd8af8112f184ffacced2a92b59d25500f6b9b. Before this, both triangles in the example would be white, and after they are both colored.

### CODE REVIEW

**Comment by phaazon** [comment]
I’m creating an issue for follow-up, and I’ll have a look on my side. Thanks for the PR! If the problem was introduced in 58fd8af, it’s likely it’s linked to the way textures are handled.

**Comment by iwikal** [comment]
I don't think the problem wasn't introduced in 58fd8af, I just noticed a change in behavior at that point. Rebasing this branch on older commits results in different erroneous behavior.

**Comment by phaazon** [comment]
Yeah I saw the code you started to modify, and clearly there’s something wrong here. I might push some commits to your branch to completely fix everything. Thanks for your time on this!

**Comment by iwikal** [comment]
By improving error handling in the example program, I discovered a simple error in the shader interface, and with that fixed it looks like it's working as it should!

**Comment by phaazon** [comment]
This is great, thanks! I have noticed a few things to optimize, so I’ll push some more commits and update the example, but overall it’s mergeable. Thanks!

File: luminance/src/backend/color_slot.rs [hunk 1]
```diff
@@ -114,12 +114,19 @@ macro_rules! impl_color_slot_tuple {
...
         C: GraphicsContext<Backend = B>, {
           Ok(
-            ($(<$pf as ColorSlot<B, D>>::reify_color_textures(ctx, size, mipmaps, sampler, framebuffer, attachment_index + 1)?),*)
+            ($(<$pf as ColorSlot<B, D>>::reify_color_textures(
+              ctx,
+              size,
+              mipmaps,
+              sampler,
+              framebuffer,
+              { attachment_index += 1; attachment_index - 1 }
```
Comment (phaazon) [inline luminance/src/backend/color_slot.rs:128]: I’m not a huge fan of this way of writing it. Can you please move the assignement before the call of `reify_color_textures`, please? :)

Comment (iwikal) [reply luminance/src/backend/color_slot.rs]: Consider it done!

**Pull request merged**